packages/transport/src/codec.js

```diff
@@ -0,0 +1,211 @@
+import * as API from '@ucanto/interface'
+
+/**
+ * @typedef {`${Lowercase<string>}/${Lowercase<string>}`|`${Lowercase<string>}/${Lowercase<string>}+${Lowercase<string>}`} ContentType
+ * @typedef {`${Lowercase<string>}/${Lowercase<string>}`|`${Lowercase<string>}/${Lowercase<string>};q=${number}.${number}`} MediaType
+ * @param {object} source
+ * @param {Record<ContentType, API.Transport.RequestDecoder>} source.decoders
+ * @param {Record<MediaType, API.Transport.ResponseEncoder>} source.encoders
+ * @returns {API.InboundCodec}
+ */
+export const inbound = source => new Inbound(source)
+
+/**
+ * @implements {API.InboundCodec}
+ */
+class Inbound {
+  /**
+   * @param {API.HTTPRequest} request
+   * @returns {API.ReceiptResult<API.InboundAcceptCodec, API.HTTPError>} transport
+   */
+  accept({ headers }) {
+    const contentType = headers['content-type'] || headers['Content-Type']
```

Comment:
http headers are defined as case insensitive. We should probably only check for `content-type`. `cOnTent-TypE` is also as valid as `headers['Content-Type']` and it seems reasonable to expect normalization at a higher layer